services/cannon/src/Cannon/Types.hs

```diff
@@ -117,6 +119,9 @@ wsenv = Cannon $ do
   r <- asks reqId
   pure $ WS.setRequestId r e
 
+stableRabbitmqConn :: Cannon (MVar (Maybe Connection))
+stableRabbitmqConn = Cannon $ asks stableRabbitmqConn_
```

Comment:
There is a limit on number of channels we can open per connection, so this solution will likely break because of that. I think it might be best to go back to separate connections per client for now because of two reasons:
- I think I misjudged that openning ~20k connections will break rabbitmq, perhaps we just need to run many nodes.
- For this we should probably implement some smart connection pooling, which would take some time and look very different from this, so perhaps we can split this work up like that.